components/amount-buttons.jsx

```diff
@@ -53,7 +60,36 @@ var AmountOtherButton = React.createClass({
       this.props.onRadioChange();
     }
   },
+  onInputChange: function(e) {
+    var inputValue = e.currentTarget.value;
+    var amount = "";
+
+    if (/^(\d)*[\,]?(\d){0,2}$/.test(inputValue)) {
```

Comment:
Right now adding a dot or comma can only happen once, and only before the last two characters, making it really easy to parse.